lib/parser/source/tree_rewriter.rb

```diff
@@ -381,21 +402,28 @@ def check_range_validity(range)
         range
       end
 
-      def enforce_policy(event)
-        return if @policy[event] == :accept
-        return unless (values = yield)
-        trigger_policy(event, **values)
+      EVENT_TO_POLICY = {
+        crossing_deletions:     :@crossing_deletions,
+        different_replacements: :@different_replacements,
+        swallowed_insertions:   :@swallowed_insertions,
+      }.freeze
+
+      def policy(event)
+        return :raise if event == :crossing_insertions
+
+        instance_variable_get(EVENT_TO_POLICY.fetch(event))
```

Comment:
let's rewrite it to `case`. The difference in terms of speed it too small but it makes it harder to follow. Also, this way you can combine it with `return :raise` above

Reply:
Yup. You're absolutely right.